### PR TITLE
support custom datax configuration 

### PR DESCRIPTION
**Pull request by simon824**
support custom datax configuration  #2102

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java [hunk 1]
```diff
@@ -192,24 +192,47 @@ private String buildDataxJsonFile()
...
+            job.put("content", buildDataxJobContentJson());
+            job.put("setting", buildDataxJobSettingJson());
+
+            JSONObject root = new JSONObject();
+            root.put("job", job);
+            root.put("core", buildDataxCoreJson());
+            json = root.toString();
+        }
+
+        logger.debug("datax job json : {}", json);
```
Comment (Jave-Chen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: I think logger.info would be easy to debug the whole task.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java [hunk 2]
```diff
@@ -192,24 +192,47 @@ private String buildDataxJsonFile()
...
 
-        JSONObject root = new JSONObject();
-        root.put("job", job);
-        root.put("core", buildDataxCoreJson());
 
-        logger.debug("datax job json : {}", root.toString());
+        if (dataXParameters.getCustomConfig() == 1){
+
+            json = dataXParameters.getJson().replaceAll("\\r\\n", "\n");
+
```
Comment (Jave-Chen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: Variables can also be used in datax job json, which will cause json to be invalid, such as
```
{
  ...
  "reader": {
    "name": "mysqlreader",
    "parameter": {
      "username": "${db_user}",
      "password": "${db_pwd}",
      "encoding": "UTF-8",
      "column": [${tbl_clms}],
      "connection": [
        {
          "jdbcUrl": [
            "jdbc:mysql://${host}/${db}?useUnicode=true&characterEncoding=utf8"
          ],
          "table": [
            ${table_list}
          ]
        }
      ]
    }
  }
  ...
}
```

Comment (simon824) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: here is about custom variables 
image

Comment (Jave-Chen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: > here is about custom variables
Sorry, I didn't see it. it's my fault.

**Pull request merged**